db/sqlglue.c

```diff
@@ -8613,11 +8613,17 @@ static int chunk_transaction(BtCursor *pCur, struct sqlclntstate *clnt,
     int need_ssl = 0;
 
     uint8_t **pIdxInsert = NULL, **pIdxDelete = NULL;
+    unsigned long long ins_keys_saved = 0ULL;
+    unsigned long long del_keys_saved = 0ULL;
 
     if (clnt->dbtran.crtchunksize >= clnt->dbtran.maxchunksize) {
 
-        /* Latch expressional index keys. We do not want them to be
-           cleaned up by the micro transaction we are about to commit. */
+        /* Latch partial index/index on expression related data. We do not want
+           them to be cleaned up by the micro transaction we are about to commit. */
+        if (gbl_partial_indexes) {
```

Comment:
Can you please change this to `if (gbl_expressions_indexes && pCur->db && pCur->db->ix_partial)`, just to be consistent with how expressional indexes are handled?